src/main/java/com/sforce/cd/apexUnit/client/utils/ApexClassFetcherUtils.java

```diff
@@ -432,5 +433,24 @@ public static void logTheFetchedApexClasses(String[] classIds) {
 			}
 		}
 	}
+	
+	/*
+	 * Returns a string array of namespace and class/trigger name
+	 * 
+	 * @param strLine - String value of user provided <NAMESPACE>.<CLASS/TRIGGER>
+	 */
+	public static Map<String, String> parseNamespaceAndName(String strLine){
+		Map<String, String> namespaceAndName = new HashMap<String, String>();
+		String [] parsedName = strLine.split("\\.");
+		if(parsedName.length==1){
+			namespaceAndName.put("name", strLine);
+		}else{
+			namespaceAndName.put("namespace", parsedName[0]);
+			namespaceAndName.put("name", parsedName[1]);
+		}
```

Comment:
@adarsh-ramakrishna-sfdc  
NamespacePrefix and Classname cannot contain dot.
Did not add logic to parsing more than dots as according to the syntax [here](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_namespace_prefix.htm)

A.B.C.D means
A is namespace
B is the Classname
C is method/attribute name
So I need only the first and second words to build the query for the API. Let me know if it does not sound right.